app/actors/hyrax/migrator/actors/file_upload_actor.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module Hyrax::Migrator::Actors
+  # Identifies the original files in the bag, uploads the content, and persists the paths in the works env
+  class FileUploadActor < Hyrax::Migrator::Actors::AbstractActor
+    aasm do
+      state :file_upload_initial, initial: true
+      state :file_upload_succeeded, :file_upload_failed
+
+      event :file_upload_initial do
+        transitions from: %i[file_upload_initial file_upload_failed],
+                    to: :file_upload_initial
+      end
+      event :file_upload_failed, after: :post_fail do
+        transitions from: :file_upload_initial,
+                    to: :file_upload_failed
+      end
+      event :file_upload_succeeded, after: :post_success do
+        transitions from: :file_upload_initial,
+                    to: :file_upload_succeeded
+      end
+    end
+
+    ##
+    # Use the FileUploadService and configurations to upload the file from the bag and persis the new paths
+    # in the works env
+    def create(work)
+      super
+      file_upload_initial
+      update_work
+      @uploaded_file = service.upload_file_content
+      @uploaded_file ? file_upload_succeeded : file_upload_failed
+    rescue StandardError => e
+      file_upload_failed
+      log("failed during file upload: #{e.message}")
+    end
+
+    private
+
+    #:nocov:
+    def service
+      @service ||= Hyrax::Migrator::Services::FileUploadService.new(@work.file_path, config)
+    end
+    #:nocov:
+
+    def post_fail
+      update_work
+    end
+
+    def post_success
+      if @uploaded_file['url'].present?
+        @work.env[:attributes].merge(remote_files: [@uploaded_file])
+      else
+        @work.env[:attributes].merge(uploaded_files: [hyrax_local_file_uploaded.id])
```

Comment:
I forgot to mention that this `hyrax_local_file_uploaded` needs to be called prior to the actor claiming success or fail.